ComponentKit/Core/ComponentUtilities.h

```diff
@@ -75,6 +75,31 @@ namespace CK {
 
     return newVector;
   }
+
+  template <typename T, typename Func, typename Predicate>
+  auto intersperse(std::vector<T> a, Func &&factory, Predicate predicate) -> std::vector<T>
+  {
+    std::vector<T> newVector;
+
+    if (a.size() < 2) {
+      return (a.size() == 1 && predicate(a.at(0))) ? a : newVector;
+    }
+
+    for (int i = 0; i < a.size() - 1; i++) {
```

Comment:
@eczarny I could do some fast enumeration but I'm going to a.size() - 1 so i figured indices was just easiest